### ISSUE TITLE
DeployAzureResourceGroup: should be possible to use "complete" mode

### ISSUE THREAD

**Issue by akunaatrium** [Microsoft/vsts-tasks#1863]
Currently the ARM templates are deployed using the Incremental mode. It should be possible to deploy using the Complete mode also.
One reason being that when a Custom Script VM extension is updated and pushed to Azure Storage when doing continuous deployment, the VM that was already created in previous deployments does not download/run the updated script. If complete mode was possible, the VM would be recreated and then the updated script is downloaded and executed and the final state of the system would be clean and up to date.

**Comment by RoopeshNair** [comment]
@akunaatrium,

We have this on our backlog. However it will be 3-4 sprints before we get to it.

**Comment by gregjhogan** [comment]
@akunaatrium, in the mean time, you are supposed to be able to force a redeployment of any VM extension by setting the `forceUpdateTag` property to a different value.  If you want it to always re-deploy and your deployment name is always unique, you could use `[deployment().name]` for the value.

**Comment by akunaatrium** [comment]
Yep, true, thanks. I actually also figured that out in the mean time :)

**Comment by gregjhogan** [comment]
I have run into multiple situations where `forceUpdateTag` doesn't cause extensions to completely re-install.  I did, however, find a way to get DeployAzureResourceGroup to do a complete mode deployment.  I discovered that any command line parameter can go into Override Template Parameters, so put -Mode Complete -Force into that parameter.

![image](https://cloud.githubusercontent.com/assets/4112046/18838832/6987a796-83ce-11e6-8878-cd2453c185c9.png)

It would also be nice if you re-opened this issue until it is actually resolved.
